Gateway/Http/Client/TransactionPayment.php

```diff
@@ -14,42 +14,60 @@
 use Adyen\AdyenException;
 use Adyen\Client;
 use Adyen\Payment\Helper\Data;
+use Adyen\Payment\Helper\GiftcardPayment;
 use Adyen\Payment\Helper\Idempotency;
+use Adyen\Payment\Helper\OrdersApi;
 use Adyen\Payment\Model\PaymentResponse;
 use Adyen\Payment\Model\PaymentResponseFactory;
 use Adyen\Payment\Model\ResourceModel\PaymentResponse as PaymentResponseResourceModel;
+use Adyen\Service\Checkout;
 use Magento\Payment\Gateway\Http\ClientInterface;
-use Adyen\Payment\Model\ApplicationInfo;
 use Magento\Payment\Gateway\Http\TransferInterface;
+use Magento\Store\Model\StoreManagerInterface;
 
 class TransactionPayment implements ClientInterface
 {
     private Data $adyenHelper;
-    private ApplicationInfo $applicationInfo;
     private PaymentResponseFactory $paymentResponseFactory;
     private PaymentResponseResourceModel $paymentResponseResourceModel;
     private Idempotency $idempotencyHelper;
+    private OrdersApi $orderApiHelper;
+    private StoreManagerInterface $storeManager;
+    private GiftcardPayment $giftcardPaymentHelper;
+
+    private ?int $remainingOrderAmount;
 
     public function __construct(
-        Data                         $adyenHelper,
-        ApplicationInfo              $applicationInfo,
-        PaymentResponseFactory       $paymentResponseFactory,
+        Data $adyenHelper,
+        PaymentResponseFactory $paymentResponseFactory,
         PaymentResponseResourceModel $paymentResponseResourceModel,
-        Idempotency                  $idempotencyHelper
+        Idempotency $idempotencyHelper,
+        OrdersApi $orderApiHelper,
+        StoreManagerInterface $storeManager,
+        GiftcardPayment $giftcardPaymentHelper
     ) {
         $this->adyenHelper = $adyenHelper;
-        $this->applicationInfo = $applicationInfo;
         $this->paymentResponseFactory = $paymentResponseFactory;
         $this->paymentResponseResourceModel = $paymentResponseResourceModel;
         $this->idempotencyHelper = $idempotencyHelper;
+        $this->orderApiHelper = $orderApiHelper;
+        $this->storeManager = $storeManager;
+        $this->giftcardPaymentHelper = $giftcardPaymentHelper;
+
+        $this->remainingOrderAmount = null;
     }
 
-    public function placeRequest(TransferInterface $transferObject): mixed
+    public function placeRequest(TransferInterface $transferObject): array
     {
         $request = $transferObject->getBody();
         $headers = $transferObject->getHeaders();
         $clientConfig = $transferObject->getClientConfig();
 
+        $client = $this->adyenHelper->initializeAdyenClient();
+        $service = $this->adyenHelper->createAdyenCheckoutService($client);
```

Comment:
Sine you are overriding those vars in lines 78:88, no need for them here
```suggestion
```

Reply:
Hello @hossam-adyen, thank you for your review. I agree!